apps/tasks/src/jobs/integrations/media-organizer.ts

```diff
@@ -0,0 +1,58 @@
+import dayjs from "dayjs";
+import SuperJSON from "superjson";
+
+import { decryptSecret } from "@homarr/common";
+import { db, eq } from "@homarr/db";
+import { items } from "@homarr/db/schema/sqlite";
+import { SonarrIntegration } from "@homarr/integrations";
+import { createCacheChannel } from "@homarr/redis";
+import type { WidgetComponentProps } from "@homarr/widgets";
+
+import { EVERY_MINUTE } from "~/lib/cron-job/constants";
+import { createCronJob } from "~/lib/cron-job/creator";
+
+export const mediaOrganizerJob = createCronJob(EVERY_MINUTE).withCallback(async () => {
+  const itemsForIntegration = await db.query.items.findMany({
+    where: eq(items.kind, "calendar"),
+    with: {
+      integrations: {
+        with: {
+          integration: {
+            with: {
+              secrets: {
+                columns: {
+                  kind: true,
+                  value: true,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  for (const itemForIntegration of itemsForIntegration) {
+    const integration = itemForIntegration.integrations[0]?.integration;
```

Comment:
This should iterate through all iterations as well

Reply:
Maybe it also makes sense to create a map of all integrations with matching Id and then to convert it back to a list of integrations. Then it will also not fetch events for one integration multiple times:

```ts
const items = [{id: 1}, {id: 2}, {id: 1}, {id: 3}, {id: 4}, {id: 3}, {id: 2}, {id: 1}, {id: 4}, {id: 5}];
const itemsMap = new Map<number, Integration>();
items.forEach(item => {
    if (itemsMap.has(item.id)) return;

    itemsMap.set(item.id, item);
});
```